### PR TITLE
SED-1366 the charts of the perforamnce view remain empty after filter change

### PR DESCRIPTION
**Pull request by lucianbaciu**

### CODE REVIEW

File: projects/step-frontend/src/lib/modules/timeseries/execution-page/execution-page.component.ts [hunk 1]
```diff
@@ -70,6 +92,30 @@ export class ExecutionPageComponent implements OnInit, OnDestroy {
...
   }
 
+  rebuildTimeSeries() {
+    this.migrationInProgress = true;
+    this.timeSeriesService
+      .rebuildTimeSeries(this.executionId)
+      .pipe(
+        pollAsyncTask(this._asyncTaskService),
+        delay(10_000) // in order for the migration to be completed as much as possible
+      )
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/execution-page/execution-page.component.ts:102]: Same as in the other PR, make sure that the delay really makes sense

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts [hunk 2]
```diff
@@ -240,16 +215,25 @@ export class PerformanceViewComponent implements OnInit, OnDestroy {
...
   reconstructAllCharts() {
-    this.findRequest.numberOfBuckets = this.calculateIdealNumberOfBuckets(
-      this.settings.startTime,
-      this.settings.endTime
-    );
+    this.findRequest = this.prepareFindRequest(this.settings); // we don't want to lose active filters
+    // this.findRequest.numberOfBuckets = this.calculateIdealNumberOfBuckets(
+    //   this.settings.startTime,
+    //   this.settings.endTime
+    // );
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts:224]: I wouldn't commit commented out code to master

File: projects/step-frontend/src/lib/modules/timeseries/ranger/ts-ranger.component.ts [hunk 3]
```diff
@@ -88,7 +95,9 @@ export class TSRangerComponent implements OnInit, AfterViewInit, OnChanges {
 
   selectRange(fromTimestamp?: number, toTimestamp?: number) {
     let select = this.transformRangeToSelect({ from: fromTimestamp, to: toTimestamp });
-
+    // select!.width = select!.width! + 100;
+    // console.log(this.getSize(), select);
+    console.log(this.uplot.bbox, select);
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/ranger/ts-ranger.component.ts:100]: I don't want to nag you all the time about debug / commented out code, but if we remove it in the PRs we have a much easier job with keeping the master clean.

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts [hunk 4]
```diff
@@ -17,31 +18,37 @@ export class MeasurementsPickerComponent implements OnInit {
...
     if (!this.contextId) {
       throw new Error('Context Id input must be present');
     }
     this.keywordsService = this.executionPageService.getContext(this.contextId).getKeywordsContext();
-    this.keywordsService.onKeywordsUpdated().subscribe((keywords) => {
-      this.keywords = keywords;
-      this.activeKeywords = 0;
-      Object.keys(keywords).forEach((key) => {
-        if (keywords[key].isVisible && keywords[key].isSelected) {
+    this.subscriptions.add(
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts:30]: Just FYI, another approach to control the subscriptions lifecycle - using the `takeUntil` operator:
```
// Define a subject in the component
private terminator$ = new Subject<unknown>();

...
// Before subscription add takeUntil
this.keywordsService
   .onKeywordsUpdated()
   .pipe(
     takeUntil(this.terminator$)
   )
  .subscribe((keywords) => {
     ...
  });

...
// In destroy hook invoke and complete the subject to terminate the susbscriptions

ngOnDestroy(): void {
    this.terminator$.next();
    this.terminator$.complete();
}
```

From my point of view such approach looks cleaner and easier to use.

File: projects/step-frontend/src/lib/modules/timeseries/ranger/ts-ranger.component.ts (hunk 3 above)
Comment (lucianbaciu) [reply projects/step-frontend/src/lib/modules/timeseries/ranger/ts-ranger.component.ts]: my fault. no worries. I will try to get used to check for console.logs before pushing

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts (hunk 4 above)
Comment (lucianbaciu) [reply projects/step-frontend/src/lib/modules/timeseries/performance-view/measurements/measurements-picker.component.ts:30]: Thank you. I also like it better. I will use this from now

**Review by neogucky — CHANGES_REQUESTED** [review]
Delay should be removed, refactoring as of Vladimirs comment + removing comments/console.logs would be great.

**Comment by lucianbaciu** [comment]
> Delay should be removed, refactoring as of Vladimirs comment + removing comments/console.logs would be great.

why remove the delay?

**Comment by neogucky** [comment]
> > Delay should be removed, refactoring as of Vladimirs comment + removing comments/console.logs would be great.
> 
> why remove the delay?

Because I don't see what purpose it has since the async function already is doing a delay (and actually repeating polls until the async tasks finishes)

**Comment by neogucky** [comment]
> 

Alright, but then we should change it in the BE so it becomes part of the async task OR create a second async task that is a super-set of the other tasks + the finishing one. I don't like to have a hardcoded delay. For now it is fine though and I will approve.

**Review by neogucky — DISMISSED** [review]

**Pull request merged**